### DISCUSSION TITLE
beginHole & endHole not working in some cases (6.5.9)

### DISCUSSION THREAD

**Discussion by kulabros** [pixijs/pixijs#10457] [General]
Hello dear guys, please, I've found that beginHole is not working as expected.

I'm drawing complex shape composed of many rounded and sharp rectangles, that are connected together, creating mask graphics object, including black and white rectangles and many white rounded rectangles / rect, that can overlap each other.

I have method to get the complex shape - graphics, and now I need holes in it.

And I need to draw many holes - rounded rectangle or rectangle holes into it, how many depends on created other sprites, I'm taking it's position and size. 

I have method to draw this holes into existing complex graphics object.
> And it's not working, there are no holes..

I've tested the draw holes method on simple rectangle, and it works ok. When I try test it on complex shape made only of rectangles, the hole graphics is added to original graphics, but not as holes. When I try it on complex shape with rounded rectangles, there no holes at all. 

I'm not using legacy pack, I'm on WebGL renderer. 

Is is possible to draw holes into any graphics shape, or is it limited?
Maybe Pixi.js v7 or v8, and pixi smooth graphics - can this be solve there? 

Thank you for reading and any help, have a nice day! :)

**Comment by kulabros** [comment]
I've resolved my issue other way, with some combination of masking.. Can't find how to delete the question... :)

**Reply by bigtimebuddy** [reply]
The holes need to be completely contained in the thing they are cutting a hole in. If the hole overlaps with the edge, it will not work.
